chat-commands.js

```diff
@@ -2479,6 +2479,59 @@ exports.commands = {
 	},
 	blacklisthelp: [`/blacklist [username], [reason] - Blacklists the user from the room you are in for a year. Requires: # & ~`],
 
+	bb: 'battleban',
+	battleban: function (target, room, user) {
+		if (!target) return this.parse('/help battleban');
+
+		let reason = this.splitTarget(target, true);
+		let targetUser = this.targetUser;
+
+		if (!targetUser) {
+			return this.errorReply(`User '${this.targetUsername}' not found.`);
+		}
+		if (!this.can('forcerename', targetUser)) return false;
+		if (targetUser.battleBanned) return this.errorReply(`User '${targetUser.name}' is already banned from battles.`);
+
+		const reasonText = reason ? ` (${reason})` : `.`;
+		const battlebanMessage = `${targetUser.name} was banned from starting new games by ${user.name}${reasonText}`;
+		this.privateModAction(`(${battlebanMessage})`);
+
+		// Notify staff room when a user is locked outside of it.
+		if (room.id !== 'staff' && Rooms('staff')) {
+			Rooms('staff').addByUser(user, "<<" + room.id + ">> " + battlebanMessage);
+		}
+
+		this.globalModlog("BATTLEBANNED", targetUser, ` by ${user.userid}${reasonText}`);
```

Comment:
Nitpicky but i think `BATTLE BAN` would be better

Reply:
It’s NAMELOCKED so I disagree

Reply:
Sorry it’s NAMELOCK, I could change it to BATTLEBAN without the space tho